qiita_pet/handlers/upload.py

```diff
@@ -44,6 +44,24 @@ def get(self, study_id):
                      raise_error=True)
         self.display_template(study_id, "")
 
+    @authenticated
+    def post(self, study_id):
+        try:
+            study = Study(int(study_id))
+        except QiitaDBUnknownIDError:
+            raise HTTPError(404, "Study %s does not exist" % study_id)
+        check_access(User(self.current_user), study, no_public=True,
+                     raise_error=True)
+
+        files_to_move = []
+        for v in self.get_arguments('files_to_erase', strip=True):
```

Comment:
Oopps, this answer was about the `trash` name, not about injections. Injections could happen but if you see it only can erase files within the study_id folder to which your user have access to.